gremlin-go/driver/strategies_test.go

```diff
@@ -0,0 +1,49 @@
+/*
+Licensed to the Apache Software Foundation (ASF) Under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+*/
+
+package gremlingo
+
+import (
+	"crypto/tls"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestReadOnlyStrategy(t *testing.T) {
+	testNoAuthUrl := getEnvOrDefaultString("GREMLIN_SERVER_URL", "ws://localhost:8182/gremlin")
+
+	t.Run("Test read with TestReadOnlyStrategy", func(t *testing.T) {
+		g := initializeGraph(t, testNoAuthUrl, &AuthInfo{}, &tls.Config{})
+
+		count, err := g.WithStrategies(ReadOnlyStrategy()).V().Count().ToList()
+		assert.Nil(t, err)
+		assert.NotNil(t, count)
+		assert.Equal(t, 1, len(count))
+		val, err := count[0].GetInt32()
+		assert.Equal(t, int32(6), val)
+	})
+
+	t.Run("Test write with TestReadOnlyStrategy", func(t *testing.T) {
+		g := initializeGraph(t, testNoAuthUrl, &AuthInfo{}, &tls.Config{})
+
+		_, err := g.WithStrategies(ReadOnlyStrategy()).AddV("person").Property("name", "foo").ToList()
+
+		assert.NotNil(t, err)
```

Comment:
```suggestion
		```suggestion
		_, promise, err := g.WithStrategies(ReadOnlyStrategy()).AddV("person").Property("name", "foo").Iterate()
		assert.NotNil(t, err)
		<-promise
```
```

Reply:
fixed in https://github.com/lyndonbauto/tinkerpop/pull/55/commits/fe5bc1697327acf9c284efd75c15af1edd7190ca